File: src/main/java/com/aries/tutorial/batch/BusinessMetricsBatch.java
package com.aries.tutorial.batch;

import com.aries.extension.data.BatchData;
import com.aries.extension.data.batch.MetricsDataAsBusiness;
import com.aries.extension.handler.BatchHandler;
import com.aries.extension.util.PropertyUtil;
import com.aries.extension.util.LogUtil;

public class BusinessMetricsBatch implements BatchHandler {
    @Override
    public boolean preHandle(long batchTime) {
        // TODO: Adding pre-processing code
        return true;
    }

    @Override
    public void process(BatchData[] batchData) {
        LogUtil.info("[BusinessMetricsBatch] - " +
                PropertyUtil.getValue("business_metrics_batch", "subject", "Unknown subject"));

        for(int i = 0; i < batchData.length; i++) {
            MetricsDataAsBusiness data = (MetricsDataAsBusiness) batchData[i];

            StringBuilder sb = new StringBuilder();
            sb.append("Domain ID : " + data.domainId + ", ");
            sb.append("Domain Name : " + data.domainName + ", ");
            sb.append("Service Count : " + data.serviceCount + ", ");
            sb.append("Max TPS : " + data.maxTps + ", ");
            sb.append("Active Service : " + data.maxTps + ", ");
            sb.append("Error Count : " + data.errorCount);

            LogUtil.info(sb.toString());
        }
    }
}

File: src/main/kotlin/com/aries/tutorial2/batch/BusinessMetricsBatch.kt
package com.aries.tutorial2.batch

import com.aries.extension.data.BatchData
import com.aries.extension.data.batch.MetricsDataAsBusiness
import com.aries.extension.handler.BatchHandler
import com.aries.extension.util.LogUtil
import com.aries.extension.util.PropertyUtil

class BusinessMetricsBatch : BatchHandler {
    override fun preHandle(batchTime: Long): Boolean {
        // TODO: Adding pre-processing code
        return true
    }

    override fun process(batchData: Array<BatchData>) {
        LogUtil.info("[BusinessMetricsBatch] - " + PropertyUtil.getValue("business_metrics_batch", "subject", "Unknown subject"))

        for (d in batchData) {
            val data = d as MetricsDataAsBusiness

            LogUtil.info("Domain ID : " + data.domainId)
            LogUtil.info("Domain Name : " + data.domainName)
            LogUtil.info("Call Count : " + data.serviceCount)
            LogUtil.info("Max TPS : " + data.maxTps)
            LogUtil.info("Active Service : " + data.activeService)
            LogUtil.info("Error Count : " + data.errorCount + "\n")
        }
    }
}
